### PR TITLE
JanusGraph support

### PR DESCRIPTION
**Pull request by amcp**

### CODE REVIEW

File: src/main/java/com/amazon/janusgraph/example/MarvelGraphFactory.java [hunk 1]
```diff
@@ -185,10 +185,10 @@ public String getComicBook() {
     }
 
     public static class BatchCommand implements Runnable {
-        final TitanGraph graph;
+        final JanusGraph graph;
         final BlockingQueue<Runnable> commands;
 
-        public BatchCommand(TitanGraph graph, BlockingQueue<Runnable> commands) {
+        public BatchCommand(JanusGraph graph, BlockingQueue<Runnable> commands) {
```
Comment (amcp) [inline src/main/java/com/amazon/janusgraph/example/MarvelGraphFactory.java]: use lombok

File: src/main/java/com/amazon/janusgraph/example/MarvelGraphFactory.java [hunk 2]
```diff
@@ -225,9 +225,9 @@ public void run() {
 
     public static class ComicBookCreationCommand implements Runnable {
         final String comicBook;
-        final TitanGraph graph;
+        final JanusGraph graph;
 
-        public ComicBookCreationCommand(final TitanGraph graph, final String comicBook) {
+        public ComicBookCreationCommand(final JanusGraph graph, final String comicBook) {
```
Comment (amcp) [inline src/main/java/com/amazon/janusgraph/example/MarvelGraphFactory.java]: use lombok

File: src/main/java/com/amazon/janusgraph/example/MarvelGraphFactory.java [hunk 3]
```diff
@@ -237,7 +237,7 @@ public void run() {
             createComicBook(graph, comicBook);
         }
 
-        private static Vertex createComicBook(TitanGraph graph, String value) {
+        private static Vertex createComicBook(JanusGraph graph, String value) {
```
Comment (amcp) [inline src/main/java/com/amazon/janusgraph/example/MarvelGraphFactory.java]: final on both args

File: src/main/java/com/amazon/janusgraph/example/MarvelGraphFactory.java [hunk 4]
```diff
@@ -253,9 +253,9 @@ private static Vertex createComicBook(TitanGraph graph, String value) {
 
     public static class CharacterCreationCommand implements Runnable {
         final String character;
-        final TitanGraph graph;
+        final JanusGraph graph;
 
-        public CharacterCreationCommand(final TitanGraph graph, final String character) {
+        public CharacterCreationCommand(final JanusGraph graph, final String character) {
```
Comment (amcp) [inline src/main/java/com/amazon/janusgraph/example/MarvelGraphFactory.java]: use lombok

File: src/main/java/com/amazon/janusgraph/example/MarvelGraphFactory.java [hunk 5]
```diff
@@ -265,7 +265,7 @@ public void run() {
             createCharacter(graph, character);
         }
 
-        private static Vertex createCharacter(TitanGraph graph, String value) {
+        private static Vertex createCharacter(JanusGraph graph, String value) {
```
Comment (amcp) [inline src/main/java/com/amazon/janusgraph/example/MarvelGraphFactory.java]: final on both args

File: src/main/java/com/amazon/janusgraph/example/MarvelGraphFactory.java [hunk 6]
```diff
@@ -281,10 +281,10 @@ private static Vertex createCharacter(TitanGraph graph, String value) {
     }
 
     public static class AppearedCommand implements Runnable {
-        final TitanGraph graph;
+        final JanusGraph graph;
         final Appeared appeared;
 
-        public AppearedCommand(final TitanGraph graph, Appeared appeared) {
+        public AppearedCommand(final JanusGraph graph, Appeared appeared) {
```
Comment (amcp) [inline src/main/java/com/amazon/janusgraph/example/MarvelGraphFactory.java]: use lombok

File: src/main/java/com/amazon/janusgraph/example/MarvelGraphFactory.java [hunk 7]
```diff
@@ -304,15 +304,15 @@ public void run() {
 
     }
 
-    protected static void processLine(final TitanGraph graph, Appeared appeared) {
+    protected static void processLine(final JanusGraph graph, Appeared appeared) {
```
Comment (amcp) [inline src/main/java/com/amazon/janusgraph/example/MarvelGraphFactory.java]: final on both

File: src/main/java/com/amazon/janusgraph/example/MarvelGraphFactory.java [hunk 8]
```diff
@@ -304,15 +304,15 @@ public void run() {
...
         long start = System.currentTimeMillis();
         process(graph, appeared);
         long end = System.currentTimeMillis();
         long time = end - start;
         REGISTRY.timer(TIMER_LINE).update(time, TimeUnit.MILLISECONDS);
     }
 
-    private static void process(TitanGraph graph, Appeared appeared) {
+    private static void process(JanusGraph graph, Appeared appeared) {
```
Comment (amcp) [inline src/main/java/com/amazon/janusgraph/example/MarvelGraphFactory.java]: ffinal on both

File: src/main/java/com/amazon/janusgraph/example/MarvelGraphFactory.java [hunk 9]
```diff
@@ -329,7 +329,7 @@ private static void process(TitanGraph graph, Appeared appeared) {
         characterVertex.addEdge(APPEARED, comicBookVertex);
     }
 
-    private static Vertex get(TitanGraph graph, String key, String value) {
+    private static Vertex get(JanusGraph graph, String key, String value) {
```
Comment (amcp) [inline src/main/java/com/amazon/janusgraph/example/MarvelGraphFactory.java]: final on both

File: src/test/java/com/amazon/janusgraph/GraphOfTheGodsFull.java [hunk 10]
```diff
@@ -29,7 +29,7 @@ public GraphOfTheGodsFull()
         super(17 /*edges*/, 12 /*vertices*/);
     }
 
-    public static void loadData(TitanGraph graph)
+    public static void loadData(JanusGraph graph)
```
Comment (amcp) [inline src/test/java/com/amazon/janusgraph/GraphOfTheGodsFull.java]: investigate why this class had to be copied and find a way to get rid of it.

File: src/test/java/com/amazon/janusgraph/MultiMarvelTest.java [hunk 11]
```diff
@@ -28,7 +28,7 @@
  */
 public class MultiMarvelTest extends AbstractMarvelTest {
 
-    private static TitanGraph GRAPH;
+    private static JanusGraph GRAPH;
```
Comment (amcp) [inline src/test/java/com/amazon/janusgraph/MultiMarvelTest.java]: why static?

File: src/test/java/com/amazon/janusgraph/TestGraphUtil.java [hunk 12]
```diff
@@ -83,20 +83,20 @@ public boolean isUnlimitedIops() {
         return unlimitedIops;
     }
 
-    public TitanGraph openGraph(BackendDataModel backendDataModel) {
+    public JanusGraph openGraph(BackendDataModel backendDataModel) {
```
Comment (amcp) [inline src/test/java/com/amazon/janusgraph/TestGraphUtil.java]: final on args

File: src/test/java/com/amazon/janusgraph/TestGraphUtil.java [hunk 13]
```diff
@@ -83,20 +83,20 @@ public boolean isUnlimitedIops() {
...
         Configuration config = createTestGraphConfig(backendDataModel);
-        TitanGraph graph = TitanFactory.open(config);
+        JanusGraph graph = JanusGraphFactory.open(config);
         return graph;
     }
 
-    public TitanGraph openGraphWithElasticSearch(BackendDataModel backendDataModel) {
+    public JanusGraph openGraphWithElasticSearch(BackendDataModel backendDataModel) {
```
Comment (amcp) [inline src/test/java/com/amazon/janusgraph/TestGraphUtil.java]: final on args

File: src/test/java/com/amazon/janusgraph/TestGraphUtil.java [hunk 14]
```diff
@@ -83,20 +83,20 @@ public boolean isUnlimitedIops() {
...
         Configuration config = createTestGraphConfig(backendDataModel);
         addElasticSearchConfig(config);
-        TitanGraph graph = TitanFactory.open(config);
+        JanusGraph graph = JanusGraphFactory.open(config);
```
Comment (amcp) [inline src/test/java/com/amazon/janusgraph/TestGraphUtil.java]: return JanusGraphFactory.open(config);

File: src/test/java/com/amazon/janusgraph/TestGraphUtil.java [hunk 15]
```diff
@@ -83,20 +83,20 @@ public boolean isUnlimitedIops() {
...
         return graph;
     }
 
-    public void tearDownGraph(TitanGraph graph) throws BackendException {
+    public void tearDownGraph(JanusGraph graph) throws BackendException {
```
Comment (amcp) [inline src/test/java/com/amazon/janusgraph/TestGraphUtil.java]: final

File: src/test/resources/dynamodb-local.properties [hunk 16]
```diff
@@ -9,10 +9,10 @@ metrics.enabled=true
 #metrics.console.interval=480000
 #metrics.csv.interval=500
 #metrics.csv.directory=metrics
-# Turn off titan retries as we batch and have our own exponential backoff strategy.
+# Turn off janusgraph retries as we batch and have our own exponential backoff strategy.
```
Comment (amcp) [inline src/test/resources/dynamodb-local.properties]: JanusGraph

File: src/test/resources/dynamodb.properties [hunk 17]
```diff
@@ -9,10 +9,10 @@ storage.buffer-size=1024
 # Required; specify logging interval in milliseconds
 #metrics.csv.interval=500
 #metrics.csv.directory=metrics
-# Turn off titan retries as we batch and have our own exponential backoff strategy.
+# Turn off janusgraph retries as we batch and have our own exponential backoff strategy.
```
Comment (amcp) [inline src/test/resources/dynamodb.properties]: JanusGraph

File: src/test/resources/gremlin-server-local.yaml [hunk 18]
```diff
@@ -25,7 +25,7 @@ channelizer: org.apache.tinkerpop.gremlin.server.channel.WebSocketChannelizer
 graphs: {
   graph: conf/gremlin-server/dynamodb-local.properties}
 plugins:
-  - aurelius.titan
+  - aurelius.janusgraph
```
Comment (amcp) [inline src/test/resources/gremlin-server-local.yaml]: janusgraph.imports

File: src/test/resources/gremlin-server.yaml [hunk 19]
```diff
@@ -25,7 +25,7 @@ channelizer: org.apache.tinkerpop.gremlin.server.channel.WebSocketChannelizer
 graphs: {
   graph: conf/gremlin-server/dynamodb.properties}
 plugins:
-  - aurelius.titan
+  - aurelius.janusgraph
```
Comment (amcp) [inline src/test/resources/gremlin-server.yaml]: janusgraph.imports

File: src/main/java/com/amazon/janusgraph/example/MarvelGraphFactory.java (hunk 1 above)
Comment (amcp) [inline src/main/java/com/amazon/janusgraph/example/MarvelGraphFactory.java]: done

File: src/main/java/com/amazon/janusgraph/example/MarvelGraphFactory.java (hunk 2 above)
Comment (amcp) [inline src/main/java/com/amazon/janusgraph/example/MarvelGraphFactory.java]: done

File: src/main/java/com/amazon/janusgraph/example/MarvelGraphFactory.java (hunk 3 above)
Comment (amcp) [inline src/main/java/com/amazon/janusgraph/example/MarvelGraphFactory.java]: done

File: src/main/java/com/amazon/janusgraph/example/MarvelGraphFactory.java (hunk 4 above)
Comment (amcp) [inline src/main/java/com/amazon/janusgraph/example/MarvelGraphFactory.java]: done

File: src/main/java/com/amazon/janusgraph/example/MarvelGraphFactory.java (hunk 5 above)
Comment (amcp) [inline src/main/java/com/amazon/janusgraph/example/MarvelGraphFactory.java]: done

File: src/main/java/com/amazon/janusgraph/example/MarvelGraphFactory.java (hunk 6 above)
Comment (amcp) [inline src/main/java/com/amazon/janusgraph/example/MarvelGraphFactory.java]: done

File: src/main/java/com/amazon/janusgraph/example/MarvelGraphFactory.java (hunk 7 above)
Comment (amcp) [inline src/main/java/com/amazon/janusgraph/example/MarvelGraphFactory.java]: done

File: src/main/java/com/amazon/janusgraph/example/MarvelGraphFactory.java (hunk 9 above)
Comment (amcp) [inline src/main/java/com/amazon/janusgraph/example/MarvelGraphFactory.java]: done

File: src/test/java/com/amazon/janusgraph/GraphOfTheGodsFull.java (hunk 10 above)
Comment (amcp) [inline src/test/java/com/amazon/janusgraph/GraphOfTheGodsFull.java]: no issue here

File: src/test/java/com/amazon/janusgraph/MultiMarvelTest.java (hunk 11 above)
Comment (amcp) [inline src/test/java/com/amazon/janusgraph/MultiMarvelTest.java]: added TODO

File: src/test/java/com/amazon/janusgraph/TestGraphUtil.java (hunk 12 above)
Comment (amcp) [inline src/test/java/com/amazon/janusgraph/TestGraphUtil.java]: done

File: src/test/java/com/amazon/janusgraph/TestGraphUtil.java (hunk 13 above)
Comment (amcp) [inline src/test/java/com/amazon/janusgraph/TestGraphUtil.java]: done

File: src/test/java/com/amazon/janusgraph/TestGraphUtil.java (hunk 14 above)
Comment (amcp) [inline src/test/java/com/amazon/janusgraph/TestGraphUtil.java]: done

File: src/test/java/com/amazon/janusgraph/TestGraphUtil.java (hunk 15 above)
Comment (amcp) [inline src/test/java/com/amazon/janusgraph/TestGraphUtil.java]: done

File: src/test/resources/dynamodb-local.properties (hunk 16 above)
Comment (amcp) [inline src/test/resources/dynamodb-local.properties]: fixed

File: src/test/resources/dynamodb.properties (hunk 17 above)
Comment (amcp) [inline src/test/resources/dynamodb.properties]: Fixed

File: src/test/resources/gremlin-server-local.yaml (hunk 18 above)
Comment (amcp) [inline src/test/resources/gremlin-server-local.yaml]: fixed

File: src/test/resources/gremlin-server.yaml (hunk 19 above)
Comment (amcp) [inline src/test/resources/gremlin-server.yaml]: fixed

**Comment by amcp** [comment]
Will merge if this build passes. https://travis-ci.org/awslabs/dynamodb-titan-storage-backend/builds/227422639

**Comment by amcp** [comment]
this build passed: https://travis-ci.org/awslabs/dynamodb-titan-storage-backend/builds/227487889, so i am merging.

**Comment by rmmeans** [comment]
🤔  Am I missing something? It looks like this was closed instead of merged to master?

**Comment by amcp** [comment]
@rmmeans please try out the latest and greatest on the master branch!

**Pull request closed without merging**